Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,20 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def has_source?()
+      !version.nil?
+    end
```

Comment:
I think this was a good idea in theory, but now that I see it, this is even more unclear, since there is no obvious relationship between missing source and version being `nil`.